src/skagent/algos/maliar.py

```diff
@@ -263,40 +306,56 @@ def maliar_training_loop(
     tolerance=1e-6,
     random_seed=None,
     simulation_steps=1,
+    epochs=250,
 ):
     """
-    block - a model definition
-    loss_function : callable((df, input_vector) -> loss vector
-    states_0_n : Grid a panel of starting states
-    parameters : dict : given parameters for the model
-
-    shock_copies: int : number of copies of shocks to include in the training set omega
-                        must match expected number of shock copies in the loss function
-                        TODO: make this better, less ad hoc
-
-    loss_function is the "empirical risk Xi^n" in MMW JME'21.
-
-    max_iterations: int
-        Number of times to perform the training loop, if there is no convergence.
-    tolerance: float
-        Convergence tolerance. Training stops when the L2 norm of parameter changes
-        is below this threshold.
-    simulation_steps : int
-        The number of time steps to simulate forward when determining the next omega set for training
-    """
+    Maliar training loop for policy network training only.
+
+    This function trains only a policy network using lifetime reward loss functions.
+    It cannot use Bellman residual loss since it does not create or track value networks.
+    For joint training of both policy and value networks, use bellman_training_loop instead.
 
-    def extract_parameters(network):
-        """Extract all parameters from the network into a flat tensor."""
-        params = []
-        for param in network.parameters():
-            params.append(param.data.view(-1))
-        return torch.cat(params) if params else torch.tensor([])
+    This implements policy-only optimization of θ₂ where:
+    - θ₂: Decision rule parameters φ(·; θ₂)
+
+    Parameters
+    ----------
+    block : DBlock
+        The model definition
+    loss_function : callable
+        Loss function for policy training with signature (decision_function, input_grid) -> loss.
+        Must be a lifetime reward loss function. Cannot use Bellman residual loss since
+        this function only trains policy networks and does not create/track value networks.
+        Created by get_estimated_discounted_lifetime_reward_loss().
```

Comment:
I see why you're going this way. But I'm not thrilled about a design where there's two methods with redundant code except for one piece. It's harder to maintain two rather than one. What design can unify these?